File: Java/src/main/java/com/ormComp/model/Flat.java
package com.ormComp.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;

import java.util.List;

@Entity
@Getter
@Table(name = "flats", schema = "rental")
public class Flat {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private long id;
    @Column(name = "description")
    private String description;
    @Column(name = "capacity")
    private short capacity;
    @Column(name = "daily_price_per_person")
    private double dailyPricePerPerson;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "building_id")
    private Building building;
    @Column(name = "flat_number")
    private String flatNumber;
    @OneToMany(mappedBy = "flat")
    private List<Reservation> reservation;

    @OneToMany(mappedBy = "flat")
    private List<FlatFacility> flatFacilities;

    @Override
    public String toString() {
        return "Flat{" +
                "id=" + id +
                ", description='" + description + '\'' +
                ", capacity=" + capacity +
                ", dailyPricePerPerson=" + dailyPricePerPerson +
                ", building=" + building +
                ", flatNumber='" + flatNumber + '\'' +
                '}';
    }
}

File: DotNet/Models/Flat.cs
using System;
using System.Collections.Generic;

namespace Endpoints.Models;

public partial class Flat
{
    public long Id { get; set; }

    public string? Description { get; set; }

    public short Capacity { get; set; }

    public float DailyPricePerPerson { get; set; }

    public long BuildingId { get; set; }

    public string? FlatNumber { get; set; }

    public virtual Building Building { get; set; } = null!;

    public virtual ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();

    public virtual ICollection<Facility> Facilities { get; set; } = new List<Facility>();
}
